### ISSUE TITLE
Consistency for Paths and URLs in config files

### ISSUE THREAD

**Issue by akberc** [ceylon/ceylon-ide-common#33]
With release 1.2.1, the repository paths in the user `.ceylon/config` files require a different syntax than before, and a different syntax to the project `.ceylon/config` file.

Requests:
1. While the global and and user config files can be OS-specific, the project config file needs to be OS-agnostic.
2. Documented correct URL and path syntax for Unix and Windows.  Right now, it is trial and error to get the right URL for the repositories.  If it is deemed relative by the IDE, for example without the `:` or lowercase drive, the IDE creates a folder structure under the project structure for the user repos.
3. Maybe, we need standard Path and URL objects in the common config so that they are read and written consistently and throw consistent exceptions if malformed.

Here is the config working for me right now. The absolute URL did not use to need the `:` :

* User config file
```
[defaults]
encoding=UTF-8
offline=false

[repository "CACHE"]
url=/D:/work/ceylon/cache

[repository "USER"]
url=/D:/work/ceylon/repo

[formattool]
profile=default
```

* Project config file
```
[compiler]
source=.\\system
source=.\\help
source=.\\server
```

**Comment by quintesse** [comment]
Yes please. The config file should be portable, you must be able to commit this in Git and not worry about others not being able to use it! 

Always use Unix conventions. 

Never voluntarily use absolute paths (you can't prevent the user from doing so but the IDE should at least never do so by itself)
